### ISSUE TITLE
Regression in 8.1.0: browser targets resolved from process.cwd() instead of per-file, breaking monorepos

### ISSUE THREAD

**Issue by Sho0226** [ismay/stylelint-no-unsupported-browser-features#311]
## Summary

Since **8.1.0**, this plugin resolves the browserslist query **once at module load time from `process.cwd()`**, instead of resolving it relative to each linted CSS file as **8.0.x** did (by delegating to `doiuse`).

In a monorepo where `.browserslistrc` files live in sub-packages (e.g. `apps/*/.browserslistrc`) but Stylelint is executed from the repository root (which is what `lint-staged` does by default), `process.cwd()` has **no** browserslist config in its ancestry. Browserslist therefore falls back to its built-in `defaults` (`> 0.5%, last 2 versions, Firefox ESR, not dead`), which includes `op_mini all` and `kaios`. This produces false positives such as `css-grid`, `css-variables`, `viewport-units`, and `flexbox-gap` being reported as "not supported by Opera Mini all", even though every project's actual target (`last 2 Chrome/Edge/Safari versions, ios >= 15`) fully supports them.

## Environment

- `stylelint-no-unsupported-browser-features`: `8.1.1` (also affects `8.1.0`)
- `stylelint`: `17.13.0`
- `browserslist`: `4.28.2`
- pnpm monorepo; `.browserslistrc` exists in each app package, not at the repo root.

## Steps to reproduce

1. A monorepo with `packages/foo/.browserslistrc` containing e.g. `last 2 Chrome versions` (something that supports CSS Grid), and **no** `.browserslistrc` at the repo root.
2. A CSS file `packages/foo/style.css` using `display: grid;`.
3. Run Stylelint **from the repo root**: `stylelint packages/foo/style.css` (this is how `lint-staged` invokes it).

### Actual (8.1.0 / 8.1.1)
style.css
  1:3  ✖  Unexpected browser feature "css-grid" is not supported by Opera Mini all  plugin/no-unsupported-browser-features
The check uses browserslist `defaults` (Opera Mini / KaiOS), ignoring `packages/foo/.browserslistrc`.

### Actual (8.0.4)
No errors — the target is resolved from the file's own `.browserslistrc`.

### Expected
Browser targets should be resolved relative to the **linted file's location** (walking up from the file), so each file uses its nearest `.browserslistrc`, regardless of the current working directory Stylelint runs from.

## Root cause

`lib/index.js`:

```js
import browserslist from 'browserslist';          // added in 8.1.0
...
const defaultBrowsers = browserslist();           // resolved ONCE from process.cwd() at module load
...
if (!doiuseOptions.browsers) {
  doiuseOptions.browsers = defaultBrowsers;        // forces that single global list into doiuse
}

In 8.0.4 doiuse was invoked without browsers, so it resolved browserslist per processed file. 8.1.0 pre-computes a single list from process.cwd() at import time and forces it, discarding per-file resolution.

Proposed fix

Resolve the query per file, using the path Stylelint provides on the rule's root, so browserslist walks up from the file rather than from process.cwd():

return (root, result) => {
  ...
  if (!doiuseOptions.browsers) {
    const from = root.source && root.source.input && root.source.input.file;
    doiuseOptions.browsers = browserslist(null, { path: from || process.cwd() });
  }
  ...
};

(Also move the resolution out of module scope so it isn't cached across files.) This restores the 8.0.x behavior while keeping the explicit browsers option working when provided.

Workarounds (for others hitting this)

- Set the browsers option explicitly in the rule config (a single static list; not per-file), or
- Add a .browserslistrc at the repo root, or
- Pin to 8.0.4.

---
